### ISSUE TITLE
Memory profiling on python

### ISSUE THREAD

**Issue by arunthamma** [pyroscope-io/pyroscope#316]
Hi,

I have started using pyroscope for profiling cloud applications developed using **python** and it's fantastic. Currently I see that we are only profiling CPU. Can I get some information on integrating memory//io profiling with pyroscope.

Regards.

**Comment by petethepig** [comment]
Yeah, we're planning to look into it at some point. There's an issue for it where we're going to track progress https://github.com/pyroscope-io/pyroscope/issues/316 but we don't have a timeline at the moment. Any pointers on how this can be implemented are very much appreciated!

**Comment by Rperry2174** [comment]
Was talking to someone who suggested we check out https://github.com/mgedmin/objgraph. Looking at there docs it seems like we could definitely turn the output into a flamegraph. 

The biggest concern is the amount of overhead that this would add to our Python integration

**Comment by nixjdm** [comment]
Hey, Python developer here. I've just been pointed to check out this project, looks great! I know of another tool I'd suggest checking out.

I used py-spy (directly) for some profiling needs a while back, but it couldn't provide all that I wanted. Eventually I found https://github.com/plasma-umass/scalene and have been using it since. It can profile memory usage. Check out the profiler comparison table they have and sample output in their README.

It doesn't auto-generate a flame graph (but it's a "good first issue" https://github.com/plasma-umass/scalene/issues/33). Like py-spy, you'd run it via CLI, which I'd think is how you're using py-spy currently. Hopefully they'll open up a [Python API](https://github.com/plasma-umass/scalene/issues/240) not long from now too.

**Comment by Rperry2174** [comment]
Adding https://github.com/bloomberg/memray as a potential solution here to be able to support memory profiling

**Comment by ducminhle** [comment]
Hi, any update? Are you waiting for Python 3.12 and develop this feature for Python 3.12 and newer versions?

**Comment by meikuam** [comment]
Hi, any update on this?

**Comment by Rperry2174** [comment]
@ducminhle still looking for suitable memory profilers out there to support ingesting data from:
https://github.com/bloomberg/memray/issues/445

**Comment by ducminhle** [comment]
Did you check https://github.com/plasma-umass/scalene, @Rperry2174? It support CPU, GPU and memory profiler.

**Comment by mrchi** [comment]
Looking forward to using this feature as soon as possible.

**Comment by yaroslavkasatikov** [comment]
dear team, 

Do you have any updates here?

I found the article about phlare https://martinheinz.dev/blog/89 (which seems to be ancestor of Pyroscope). They discussed about python memory and shared this screenshot - 
![image](https://github.com/grafana/pyroscope/assets/7746585/987d0942-4a91-4a35-9a01-fa74b111899d)
)

And that's that I could find related to python memory continues profiling.

May be anyone knows any other solution? Will be appreciative for any advice :(

**Comment by korniltsev** [comment]
Have you tried pypprof ?  I've never tried it, but according to the project description it should work.

**Comment by yaroslavkasatikov** [comment]
Hey @korniltsev 

Yes, I have tried it and it works fine with one detail: you should use Python 3.11 for it.
According the article I have shared, I can tell you that it works with Pyroscope too.

So it can be great if you will try to take pypprof profiler for memory profiling in mainstream.
There is the sad new here.. Pypprof last commit was done several ages ago:)

**Comment by jcarbelbide** [comment]
Hello, just started using Pyroscope and I'm really liking it! I'm using it with Python, and just found that memory profiling isn't yet supported with Python, which led me to this issue. Is there any update on this? Any chance memory profiling will be added in the future?

**Comment by pythonwood** [comment]
I just done before:
step 1
docker run --name parcademo-python-pprof -p 0.0.0.0:8081:8081 blakaneyk/parcademo:python-pprof-3-11-1
step 2
alloy scrapy and push pyroscope
> 
> 或许有人知道其他解决方法？非常感谢任何建议 :(



> dear team,
> 
> Do you have any updates here?
> 
> I found the article about phlare https://martinheinz.dev/blog/89 (which seems to be ancestor of Pyroscope). They discussed about python memory and shared this screenshot - ![image](https://github.com/grafana/pyroscope/assets/7746585/987d0942-4a91-4a35-9a01-fa74b111899d) )
> 
> And that's that I could find related to python memory continues profiling.
> 
> May be anyone knows any other solution? Will be appreciative for any advice :(

**Comment by msaeedevops** [comment]
Hi Is there any update on supporting memory for python in pyroscope

**Comment by tiffanyfay** [comment]
https://grafana.com/docs/pyroscope/latest/configure-client/language-sdks/python/#configure-memory-profiling is available now.

**Comment by korniltsev-grafanista** [comment]
https://github.com/grafana/pyroscope-python/pull/83
